storage/s3.go

```diff
@@ -1160,6 +1160,71 @@ func (s *S3) GetBucketVersioning(ctx context.Context, bucket string) (string, er
 
 }
 
+func (s *S3) HeadBucket(ctx context.Context, url *url.URL) (*Bucket, error) {
+
+	output, err := s.api.HeadBucketWithContext(ctx, &s3.HeadBucketInput{
+		Bucket: aws.String(url.Bucket),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil {
+		return nil, errors.New("output is nil")
+	}
+
+	return &Bucket{
+		Name: url.Bucket,
+	}, nil
+
+}
+
+func (s *S3) HeadObject(ctx context.Context, url *url.URL) (*Object, map[string]string, error) {
+
+	input := &s3.HeadObjectInput{
+		Bucket:       aws.String(url.Bucket),
+		Key:          aws.String(url.Path),
+		RequestPayer: s.RequestPayer(),
+	}
+	if url.VersionID != "" {
+		input.SetVersionId(url.VersionID)
+	}
+
+	output, err := s.api.HeadObjectWithContext(ctx, input)
+	if err != nil {
+		if errHasCode(err, "NotFound") {
+			return nil, nil, &ErrGivenObjectNotFound{ObjectAbsPath: url.Absolute()}
+		}
+		return nil, nil, err
+	}
+
+	var storageClassStr string
+
+	// There is a bug or a feature that causes the StorageClass to be nil
+	// when the object is created with the default storage class.
+	// In this case, we set the storage class to STANDARD.
+
+	if output.StorageClass != nil {
+		storageClassStr = aws.StringValue(output.StorageClass)
+	} else {
+		storageClassStr = "STANDART"
+	}
```

Comment:
this behaviour is documented;
> https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html#AmazonS3-HeadObject-response-header-StorageClass
> Provides storage class information of the object. Amazon S3 returns this header for all objects **except for S3 Standard storage class** objects.

you can add a link to docs instead of this comment.

something like below could be easier to read;
```go
	storageClass := "STANDARD"
	if output.StorageClass != nil {
		storageClass = aws.StringValue(output.StorageClass)
	}

```